src/lir/lir_type.rs

```diff
@@ -0,0 +1,157 @@
+            ast::StructType::UserDefined(i) => StructType::UserDefined(*i),
+        }
+    }
+}
+
+impl fmt::Display for StructType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            StructType::UserDefined(i) => write!(f, "{}", i),
+            StructType::SolAccountInfo => write!(f, "SolAccountInfo"),
+            StructType::SolAccountMeta => write!(f, "SolAccountMeta"),
+            StructType::ExternalFunction => write!(f, "ExternalFunction"),
+            StructType::SolParameters => write!(f, "SolParameters"),
+            StructType::Vector(elem_ty) => write!(f, "vector<{}>", elem_ty),
+        }
+    }
+}
+
+impl fmt::Display for LIRType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.lir_type)
+    }
+}
+
+impl fmt::Display for Type {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Type::Bool => write!(f, "bool"),
+            Type::Int(width) => write!(f, "int{}", width),
+            Type::Uint(width) => write!(f, "uint{}", width),
+            Type::Bytes(width) => write!(f, "bytes{}", width),
+            Type::Ptr(ty) => write!(f, "ptr<{}>", ty),
+            Type::StoragePtr(immutable, ty) => {
+                if *immutable {
+                    write!(f, "const_storage_ptr<{}>", ty)
+                } else {
+                    write!(f, "storage_ptr<{}>", ty)
+                }
+            }
+            Type::Array(ty, len) => {
+                write!(f, "{}", ty)?;
+                len.iter().for_each(|len| match len {
+                    ArrayLength::Fixed(len) => write!(f, "[{}]", len).unwrap(),
+                    ArrayLength::Dynamic => write!(f, "[]").unwrap(),
+                    ArrayLength::AnyFixed => write!(f, "[?]").unwrap(),
+                });
+                Ok(())
+            }
+            Type::Slice(ty) => write!(f, "slice<{}>", ty),
+            Type::Struct(ty) => write!(f, "struct.{}", ty),
+            Type::Function { params, returns } => {
+                write!(f, "fn(")?;
+                for (i, param) in params.iter().enumerate() {
+                    if i != 0 {
+                        write!(f, ", ")?;
+                    }
+                    write!(f, "{}", param)?;
+                }
+                write!(f, ")")?;
+                if !returns.is_empty() {
+                    write!(f, " -> (")?;
+                    for (i, ret) in returns.iter().enumerate() {
+                        if i != 0 {
+                            write!(f, ", ")?;
+                        }
+                        write!(f, "{}", ret)?;
+                    }
+                    write!(f, ")")?;
+                } else {
+                    write!(f, " -> ()")?;
+                }
+                Ok(())
+            }
+            Type::Mapping { key_ty, value_ty } => {
+                write!(f, "mapping<{} -> {}>", key_ty, value_ty)
```

Comment:
Solidity already has syntax for describing these types. I don't think inventing new syntax is an improvement.

Reply:
Hi Sean, I updated this with Solidity mapping syntax instead.